### PR TITLE
parallelize image rotations

### PR DESCRIPTION
**Pull request by Vermeille**
Hello Davis,

This PR is about parallelizing `add_image_rotations` which takes an unreasonable amount of time on my machine, while there are a lot of idling CPUs.

I'm not sure about some low level arbitrary choices I've made and implementation details, so please, get back to me. I'll remove the debug prints and all that while treating your feedback.

Many thanks!

### CODE REVIEW

**Comment by davisking** [comment]
Thanks for the PR. 

This should probably use dlib::parallel_for().  And yes, please remove unrelated things like print statements and other stuff.  A PR should be focused and clean.

**Comment by Vermeille** [comment]
Hello again,

I think that fixes what you asked for :)

Hope it helps!

File: dlib/image_transforms/interpolation.h [hunk 1]
```diff
@@ -1543,33 +1546,32 @@ namespace dlib
...
+        using namespace impl;
 
-        std::vector<T> objtemp;
-        std::vector<U> objtemp2;
-        typename image_array_type::value_type temp;
-        for (long i = 0; i < angles.size(); ++i)
-        {
-            for (unsigned long j = 0; j < images.size(); ++j)
+        image_array_type new_images(images.size() * angles.size());
+        Objects new_objects(images.size() * angles.size());
```
Comment (davisking) [inline dlib/image_transforms/interpolation.h]: Don't make the Objects typedef. You only use it once.  So adding it is just more code and more confusing indirection for someone reading it.  Just say std::vector<std::vector<T>>.  That's much clearer.

File: dlib/image_transforms/interpolation.h [hunk 2]
```diff
@@ -1543,33 +1546,32 @@ namespace dlib
...
+        Objects2 new_objects2(images.size() * angles.size());
+
+        dlib::parallel_for(0, images.size(), [&](int j) {
+            typename image_array_type::value_type temp;
+
+            int dst_base = j * angles.size();
+            for (long i = 0; i < angles.size(); ++i)
             {
+                int dst = dst_base + i;
```
Comment (davisking) [inline dlib/image_transforms/interpolation.h]: Be consistent in index type.  In particular, these ints should be longs.

File: dlib/image_transforms/interpolation.h (hunk 1 above)
Comment (Vermeille) [reply dlib/image_transforms/interpolation.h]: Agreed. Fixed.

File: dlib/image_transforms/interpolation.h (hunk 2 above)
Comment (Vermeille) [reply dlib/image_transforms/interpolation.h]: I guess it's right, now?

**Comment by davisking** [comment]
Looks good, thanks for the PR :)

**Pull request merged**